moabb/evaluations/splitters.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+from sklearn.model_selection import (
+    BaseCrossValidator,
+    GroupKFold,
+    LeaveOneGroupOut,
+    StratifiedKFold,
+)
+
+
+class WithinSubjectSplitter(BaseCrossValidator):
+
+    def __init__(self, n_folds):
+        self.n_folds = n_folds
+
+    def get_n_splits(self, metadata):
+        sessions_subjects = len(metadata.groupby(["subject", "session"]).first())
+        return self.n_folds * sessions_subjects
+
+    def split(self, X, y, metadata, **kwargs):
+
+        subjects = metadata.subject.values
+
+        split = IndividualWithinSubjectSplitter(self.n_folds)
+
+        for subject in np.unique(subjects):
+
+            X_, y_, meta_ = (
+                X[subjects == subject],
+                y[subjects == subject],
+                metadata[subjects == subject],
+            )
+
+            yield split.split(X_, y_, meta_)
+
+
+class IndividualWithinSubjectSplitter(BaseCrossValidator):
+
+    def __init__(self, n_folds):
+        self.n_folds = n_folds
+
+    def get_n_splits(self, metadata):
+        return self.n_folds
+
+    def split(self, X, y, metadata, **kwargs):
+
```

Comment:
```suggestion
        assert len(np.unique(metadata.subject)) == 1
```
Just to be sure the class is used in the right way